### PR TITLE
Bug fix for creating a volume

### PR DESCRIPTION
**Pull request by sonicaj**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/zfs.py [hunk 1]
```diff
@@ -717,7 +717,9 @@ def check_key(self, id, options):
     )
     def unload_key(self, id, options):
         force = options.pop('force_umount')
-        if options.pop('umount') and self.middleware.call_sync('zfs.dataset.get_instance', id)['mountpoint']:
+        if options.pop('umount') and self.middleware.call_sync(
+            'zfs.dataset.query', [['id', '=', id]], {'extra': {'retrieve_children': False}}
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/zfs.py:721]: Can we allow passing extra options to `get_instance` instead?

File: src/middlewared/middlewared/plugins/pool.py [hunk 2]
```diff
@@ -2905,18 +2903,18 @@ def transform(dataset):
         ]),
         Bool('sparse'),
         Bool('force_size'),
-        Inheritable(Str('comments')),
-        Inheritable(Str('sync', enum=['STANDARD', 'ALWAYS', 'DISABLED'])),
-        Inheritable(Str('compression', enum=ZFS_COMPRESSION_ALGORITHM_CHOICES)),
+        Inheritable(Str('comments'), default='INHERIT'),
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/pool.py:2906]: How was it crashing without `default`?

**Review by themylogin — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/pool.py [hunk 3]
```diff
@@ -2905,18 +2903,18 @@ def transform(dataset):
...
+        Inheritable(Int('quota_warning', validators=[Range(0, 100)]), default='INHERIT'),
+        Inheritable(Int('quota_critical', validators=[Range(0, 100)]), default='INHERIT'),
         Int('refquota', null=True, validators=[Or(Range(min=1024**3), Exact(0))]),
-        Inheritable(Int('refquota_warning', validators=[Range(0, 100)])),
-        Inheritable(Int('refquota_critical', validators=[Range(0, 100)])),
+        Inheritable(Int('refquota_warning', validators=[Range(0, 100)]), default='INHERIT'),
+        Inheritable(Int('refquota_critical', validators=[Range(0, 100)]), default='INHERIT'),
         Int('reservation'),
         Int('refreservation'),
         Inheritable(Int('special_small_block_size')),
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/pool.py:2920]: Don't you also need `default` here?

**Review by themylogin — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/pool.py (hunk 3 above)
Comment (sonicaj) [reply src/middlewared/middlewared/plugins/pool.py:2920]: No, this property is not valid for volumes

Comment (themylogin) [reply src/middlewared/middlewared/plugins/pool.py:2920]: So it works without `default`? Why do we need it then?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/pool.py:2920]: `INHERIT` is a no-op, zfs create api will inherit if it's not specified explicitly. It's just been defined to be explicit rather then implicit.

**Review by themylogin — APPROVED** [review]

**Pull request merged**